scripts/cleanup_edomain_templates.py

```diff
@@ -0,0 +1,221 @@
+        print "RHEVM: No old templates to delete in {}".format(edomain)
+
+    for delete_template in delete_templates[0:5]:
+        thread = Thread(target=delete_edomain_templates,
+                        args=(api, delete_template, edomain))
+        thread.daemon = True
+        thread_queue.append(thread)
+        thread.start()
+
+    for thread in thread_queue:
+        thread.join()
+
+
+def make_kwargs(args, cfme_data, **kwargs):
+    """Assembles all the parameters in case of running as a standalone script.
+       Makes sure, that the parameters given by command-line arguments
+       have higher priority.Makes sure, that all the needed parameters
+       have proper values.
+
+    Args:
+        args: Arguments given from command-line.
+        cfme_data: Data in cfme_data.yaml
+        kwargs: Kwargs generated from
+        cfme_data['template_upload']['template_upload_rhevm']
+    """
+    args_kwargs = dict(args._get_kwargs())
+
+    if len(kwargs) is 0:
+        return args_kwargs
+
+    template_name = kwargs.get('template_name', None)
+    if template_name is None:
+        template_name = cfme_data['basic_info']['appliance_template']
+        kwargs.update({'template_name': template_name})
+
+    for kkey, kval in kwargs.iteritems():
+        for akey, aval in args_kwargs.iteritems():
+            if aval is not None:
+                if kkey == akey:
+                    if kval != aval:
+                        kwargs[akey] = aval
+
+    for akey, aval in args_kwargs.iteritems():
+        if akey not in kwargs.iterkeys():
+            kwargs[akey] = aval
+
+    return kwargs
+
+
+def run(**kwargs):
+    """Calls the functions needed to cleanup templates on RHEVM providers.
+       This is called either by template_upload_all script, or by main
+       function.
+
+    Args:
+        **kwargs: Kwargs generated from
+        cfme_data['template_upload']['template_upload_rhevm'].
+    """
+    providers = cfme_data['management_systems']
+    for provider in providers:
+        if cfme_data['management_systems'][provider]['type'] != 'rhevm':
+            continue
+        if args.provider:
+            if args.provider != provider:
+                continue
+        mgmt_sys = cfme_data['management_systems'][provider]
+        edomain = mgmt_sys['template_upload']['edomain']
+        if args.edomain:
+            edomain = args.edomain
+        rhevurl = mgmt_sys['hostname']
+        rhevm_credentials = mgmt_sys['credentials']
+        username = credentials[rhevm_credentials]['username']
+        password = credentials[rhevm_credentials]['password']
+        ssh_rhevm_creds = mgmt_sys['hosts'][0]['credentials']
+        sshname = credentials[ssh_rhevm_creds]['username']
+        sshpass = credentials[ssh_rhevm_creds]['password']
+
+        apiurl = 'https://%s:443/api' % rhevurl
+
+        api = API(url=apiurl, username=username, password=password,
+                  insecure=True, persistent_auth=False)
```

Comment:
makes sense to re use the existing, providers module. changes will be made in next PR.